api/v1beta1/nutanixcluster_types.go

```diff
@@ -64,7 +64,26 @@ type NutanixClusterSpec struct {
 	// +listType=map
 	// +listMapKey=name
 	// +optional
+	//
+	// Deprecated: This field is replaced by the field controlPlaneFailureDomains and will be removed in the next apiVersion.
+	//
 	FailureDomains []NutanixFailureDomain `json:"failureDomains"`
```

Comment:
Please add omitempty to the tag (i.e. json:"failureDomains,omitempty") so that an empty slice doesn’t produce [] in the manifest when no domains are configured.

Reply:
added